Dockerfile

```diff
@@ -7,7 +7,13 @@ USER root
 # Required to build in --offline mode
 ENV CARGO_HOME=/src/.cargo
 
+# Add brupop source
 ADD ./ /src/
+
+# Ensure cargo dependencies are fetched and available in the Docker context
+RUN cargo fetch --locked --manifest-path /src/Cargo.toml
```

Comment:
Fetches sources in docker image to ensure we have necessary upstream code to build locally. This is still dependent on the Cargo.lock, so nothing new should be pulled in here. Note: this requires network access